### PR TITLE
Add sent, open & click tracking

### PR DESCRIPTION
**Pull request by codingjoe**
Watching I See You GIF

### CODE REVIEW

**Review by amureki — CHANGES_REQUESTED** [review]
Awesome feature!

I have first comments here, but I still need to do the testing part.

**Review by amureki — CHANGES_REQUESTED** [review]
Also, as you mentioned - we need to think about rate limiting.

File: README.md [hunk 1]
```diff
@@ -116,6 +117,62 @@ class MyMessage(MarkdownEmail):
...
 ### Tracking
 
+#### Sent, Open & Click Tracking
+
+Django eMark comes with built-in tracking for sent, open and click events.
+The tracking is done via a tracking pixel and a redirect view.
+
+As an added bonus, this feature also comes with an open-in-browser link, that
+allows the user to view the email in their browser, if their email client does
+not support HTML emails.
```
[the remark this answers is not in the document]
Comment (codingjoe) [reply README.md:127]: Why you hating my commas ;)

**Review by amureki — APPROVED** [review]
Shall we do a beta release first?

**Review by amureki — CHANGES_REQUESTED** [review]
I did some tests, found a couple of things. Also, tracking is questionable, we'd need to test how it works in comparison to the mailing systems like Brevo, as here are the results from my test:

- checking an email via gmail (no ad-blockers) in browser does not count as opening
- gmail on iOS also does not count at all
- Microsoft Windows 10 Mail does not count at all
- MacOS Mail.app open counts as 1 open (no matter how many times you will open it)

File: emark/models.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+    class Meta:
+        abstract = True
+
+
+class Click(ClientTrackingModelMixin):
+    """Record of a click on a link in an email."""
+
+    email = models.ForeignKey(Send, on_delete=models.CASCADE)
+    redirect_url = models.URLField()
```
Comment (amureki) [inline emark/models.py:66]: I managed to break this.
`URLField()` by default has max_length=200 but the link I had in my email is much longer:

<details>

```
status=500 method=GET path="/emark/c6f4b2f2-62f9-4b59-9c29-3b5100d48c6d/click?url=https%3A%2F%2Fvoiio-platfo-emark-trac-munpyq.herokuapp.com%2Faccounts%2Flogin%2FINq%3A%3A1qMP4r%3AZfeHkYAT9UsgjZ_mLo0gxIhQaQy3UfrJR3cpMBEcVgA%3Futm_source%3Dplatform%26utm_medium%3Demail%26utm_campaign%3DLOGIN_EMAIL%26next%3D%252Fdiscover%252F"


django.db.utils.DataError: value too long for type character varying(200)
```

</details>

**Review by amureki — APPROVED** [review]
@codingjoe switching to `` worked, opens are tracked in gmail desktop, as well as in the other clients.

So, from my side this is good to go now!

**Pull request merged**